Sources/NIOCore/NIODecodedAsyncSequence.swift

```diff
@@ -44,8 +44,91 @@ extension AsyncSequence where Element == ByteBuffer {
             maximumBufferSize: maximumBufferSize
         )
     }
+
+    /// Returns the longest possible subsequences of the sequence, in order, that
+    /// don't contain elements satisfying the given predicate. Elements that are
+    /// used to split the sequence are not returned as part of any subsequence.
+    ///
+    /// Similar to standard library's `String.split(maxSplits:omittingEmptySubsequences:whereSeparator:)`.
+    ///
+    /// Usage:
+    /// ```swift
+    /// let baseSequence = MyAsyncSequence<ByteBuffer>(...)
+    /// let splitSequence = baseSequence.split(whereSeparator: { $0 == UInt8(ascii: " ") })
+    ///
+    /// for try await buffer in splitSequence {
+    ///     print("Split by whitespaces!\n", buffer.hexDump(format: .detailed))
+    /// }
+    /// ```
+    ///
+    /// - Parameters:
+    ///   - omittingEmptySubsequences: If `false`, an empty subsequence is
+    ///     returned in the result for each pair of consecutive elements
+    ///     satisfying the `isSeparator` predicate and for each element at the
+    ///     start or end of the sequence satisfying the `isSeparator` predicate.
+    ///     If `true`, only nonempty subsequences are returned. The default
+    ///     value is `true`.
+    ///   - isSeparator: A closure that returns `true` if its argument should be
+    ///     used to split the file's bytes; otherwise, `false`.
+    /// - Returns: An `AsyncSequence` of ``ByteBuffer``s, split from the this async sequence's bytes.
+    ///
+    /// - Complexity: O(*n*), where *n* is the length of the file.
+    @inlinable
+    public func split(
```

Comment:
Is this the API you actually want? I'd think the more useful one would be getting strings? `split` is probably _too_ generic.

Reply:
I sorta think it's fine to do both `splitLines`, the challenge is in the names. The `String` one also _sort of_ implies it should be unicode-aware, which makes me a touch nervous.

Reply:
I reworked the PR to not introduce `split(separator:)` APIs.

Best I could come up with was `splitLines` and `splitLineSlices`. I was also pleasantly surprised to find out `AsyncMapSequence` is coming from std-lib, not e.g. `swift-async-algorithms`.

Right now the remaining thing to do is to add static-end-result tests assuming you-all think that's needed.